insights/parsers/ssh_client_config.py

```diff
@@ -0,0 +1,147 @@
+"""
+SshConfig - file for ``ssh client config``
+==========================================
+
+This module contains parsers that check the ssh client config files.
+
+Parsers provided by this module are:
+
+EtcSshConfig - file ``/etc/ssh/ssh_config``
+-------------------------------------------
+
+ForemanSshConfig - file ``/usr/share/foreman/.ssh/ssh_config``
+--------------------------------------------------------------
+
+"""
+from collections import namedtuple
+
+from insights.specs import Specs
+
+from .. import Parser, get_active_lines, parser
+
+
+class SshClientConfig(Parser):
+    """Base class for sshclient config file."""
+
+    def __init__(self, context):
+        self.global_lines = []
+        self.host_lines = {}
+        super(SshClientConfig, self).__init__(context)
+
+    def _split_content(self, content):
+        index = [i for i, l in enumerate(get_active_lines(content)) if l.startswith('Host ')]
```

Comment:
Every time calling the `get_active_lines`, it will iterate the whole `content` and do the same thing to each line. To get a good efficiency, better assign its result to a local variable just before the first calling.